packages/eslint-plugin/docs/rules/TEMPLATE.md

```diff
@@ -1,7 +1,3 @@
-# `your-rule-name`
```

Comment:
@armano2 starting a thread for https://github.com/typescript-eslint/typescript-eslint/pull/5249#issuecomment-1166529939: 

> > I think the idea is that the docs should not be viewed on GitHub but should be as maintainable as possible.
>
> if that is a case than we should move it out of eslint-plugin to docs

I think I agree, it would be best long term to have these docs files in the docs area. But: the old .md files still have much better SEO than the site. We can't move them all-up just yet. The most we could do would be to leave a .md file in the old location with just the redirect notice.

What would you prefer I do in this PR? I don't have strong feelings one way or another, as long as we're moving towards automating more of the rule docs pages.

Reply:
Personally - I think it's still a good idea to co-locate them with the plugin - that way they're easy to find and update for contributors.

Reply:
I am 100% behind treating the .md files in the repo as source code for most things (i.e. users shouldn't read them).
[It's what I said a year ago with the first versions of the website!](https://github.com/typescript-eslint/typescript-eslint/pull/3147#issuecomment-811506856)

Also LMAO I had a different opinion back then about where to put the files:
> TBH - I also don't see any tangible value in co-locating the docs with the packages.

So I'll say - I don't mind either way. As long as we keep things clearly documented for contributors (i.e. tests to fail if they've not been added, or not been deleted) - then it shouldn't matter.